### PR TITLE
fix(web/apply): never hand the planner a ref to a submit control, and guard every action by element facts

### LINKED ISSUE #3833: web/apply: the drive loop can click a submit control (icon-only or non-English label)
### What is wrong

In the agentic drive loop (`web/src/lib/apply/drive.ts`, at fad070fe), the guard that stops the
loop from clicking a submit control tests a different string from the one the planner is shown,
so a control the planner sees as "Submit application" can pass the guard as an empty string.

The snapshot builds the label the planner reasons over at `drive.ts:48`:

```ts
const label = clean(el.getAttribute("aria-label") || placeholder || el.textContent || value || name);
```

The click guard builds its own string at `drive.ts:192-193`:

```ts
const txt = (await loc.innerText().catch(() => "")) || (await loc.getAttribute("value").catch(() => "")) || "";
if (SUBMIT_RX.test(txt)) {
```

The two disagree in two ways, and each one is a way past the guard:

1. **The accessible name is not read.** An icon-only `<button aria-label="Submit application">`
   has no inner text and no `value`, so `txt` is `""` and `SUBMIT_RX.test("")` is false. The
   planner, meanwhile, was shown `[e12] button "Submit application"` and can name it.
2. **`SUBMIT_RX` is English and Spanish only** (`drive.ts:24`:
   `submit|send application|finish|complete application|apply and submit|enviar|finalizar`). A
   button reading "Absenden", "Envoyer", "Invia", "Verzenden" or "Отправить" does not match. The
   pattern is also `\b`-anchored, and JS word boundaries are ASCII-only, so a Cyrillic label
   cannot match even if the word is added to the list.

Either way the click executes at `drive.ts:199`.

There is a third, smaller problem behind both: the block that was meant to keep submit inputs out
of the actionable set is empty (`drive.ts:42-44`).

```ts
if (tag === "input" && ["hidden", "submit", "button", "image", "reset"].includes(itype)) {
  // surface submit buttons in the snapshot as read-only context (not actionable)
}
```

There is no `continue`, so execution falls through to `:45-51` and the submit input gets a
`data-co-ref` and an ordinary snapshot line like every other element. The comment describes an
intention the code does not carry out, which is why the click guard is currently the only layer.

A fourth path bypasses the guard without needing either label trick: the guard runs only in the
`click` branch, while the locator at `drive.ts:190` is built for any action carrying a ref. A
`type` action naming a submit control reaches `drive.ts:203-206`, where `fill()` on a button throws
and the fallback calls `loc.click()` on the same element.

### Why it matters

`goal: "full"` is reachable from the UI (`web/src/components/apply/apply-provider.tsx:358`, with a
16-turn budget at `web/src/app/api/apply/drive/route.ts:42`), so the loop is asked to work through
the form rather than only reach it. On any of these paths an application is sent with no human
review.

That is the one rule the project states without an exception: `web/AGENTS.md:20-21` ("Nothing is
ever submitted automatically ... There is no exception, no flag") and the "Ethical Use" section of
the root `AGENTS.md`. The file's own header comment says NEVER-SUBMIT is by construction, which is
the right design; the construction is just not in place yet.

### Fix shape

1. Move the decision into a pure module (`web/src/lib/apply/submit-guard.mjs`, the pattern already
   used by `exit.mjs`) exporting `snapshotLabel(facts)`, `SUBMIT_RX` and `isSubmitControl(facts)`,
   so the label the planner sees and the string the guard tests are the same function, and every
   branch is reachable from `node --test`.
2. Decide on element facts rather than on rendered text: the element's own type, HTML's default
   (a `<button>` inside a `<form>` with no honoured type is a submit button), the label, and the
   absence of a label (an icon-only control inside a form cannot be told apart from the submit
   button, so it is refused).
3. Widen `SUBMIT_RX` to the languages the portals ship, with `\p{L}` boundaries under the `u` flag
   instead of `\b`.
4. Build the snapshot lines in Node so a submit control is listed **without a ref**
   (`[--] submit "Label" (not actionable, the human submits)`). The planner can see that the
   control exists and stop, but it has no name for it, so no action can address it. Keep the click
   guard as the second layer, re-reading the element's facts at click time, since the DOM can
   change between the snapshot and the click.
5. Apply that second layer before the locator is built, so it covers every action carrying a ref
   rather than `click` alone.

I have this implemented and tested and will open a PR against this issue.

### PR DESCRIPTION
**Pull request by nikitacometa**
## What does this PR do?

The agentic drive loop's submit guard tested a string the planner never saw, so an icon-only `<button aria-label="Submit application">` (empty `innerText`) and any non-English label ("Absenden", "Envoyer", "Отправить") got past it and were clicked. This makes the refusal decide on element facts instead of rendered text, stops handing the planner a ref for a submit control in the first place, and binds the check and the action to one element handle so the node that was cleared is the node that is acted on.

## Related issue

Closes #3833

## Type of change

- [x] Bug fix
- [ ] New feature
- [ ] Documentation / translation
- [ ] Refactor (no behavior change)

## Checklist

- [x] I have read [CONTRIBUTING.md](https://github.com/career-ops-hq/career-ops/blob/main/CONTRIBUTING.md)
- [x] If this is a new feature or architecture change, I opened an issue first (bug fixes, providers, docs & translations are exempt, send those straight in)
- [x] My PR does not include personal data (CV, email, real names, scan results, or pipeline data)
- [x] I ran `node test-all.mjs` and all tests pass
- [x] My changes respect the [Data Contract](https://github.com/career-ops-hq/career-ops/blob/main/DATA_CONTRACT.md) (no modifications to user-layer files)
- [x] My changes align with the [project roadmap](https://github.com/career-ops-hq/career-ops/discussions/156)

## How it works

**New: `web/src/lib/apply/submit-guard.mjs`** (plain `.mjs`, the `exit.mjs` pattern, so `node --test` imports it with no build step).

- `snapshotLabel(facts)` is the single definition of the label the planner is shown (`aria || placeholder || text || value || name`, whitespace collapsed, 80 chars). The snapshot prints it and the guard reads the same facts, so the two cannot disagree about one element. That was the whole bug: the snapshot read `aria-label`, the guard read `innerText`.
- `isSubmitControl(facts)` refuses on four independent readings, because each alone has a blind spot: the element's own type (`input[type=submit|image]`, `button[type=submit]`); HTML's default (a `<button>` inside a `<form>` with no honoured type submits it); the wording; and the absence of any wording (an icon-only clickable inside a form is exactly the shape of the submit button we cannot identify, so it is refused rather than guessed at). `role="button"` counts as a button for the wording and silence rules but not for the typeless default, since a `<div>` does not submit on its own.
- Every label source (`aria`, `placeholder`, `text`, `value`, `title`, `alt`) is tested on its own, deduplicated, not through `labelSource`'s single winner: an `input[type="button"]` with `aria="Continue"` and `value="Submit application"` has no `text` to fall back to, so testing only the winning source would hide the wording from the guard. `name` is left out on purpose: it is a machine identifier, so `name="submit_note"` names a text box, not a submit action. `aria` is the accessible name, resolved from `aria-labelledby` in the page when there is no `aria-label`. The wording check is held to clickable controls, so a text field whose placeholder reads "Submit your feedback" stays fillable.
- Two vocabularies, because the same verb means two things on a job portal. `SUBMIT_RX` names the action that sends the application (en, es, de, fr, it, pt, nl, pl, ru, tr) and is refused wherever it appears; «Откликнуться» is in it because on hh.ru that button sends the response in one click, and `Başvur` for the same reason on kariyer.net. The bare send verbs of nl and tr (`versturen`, `verzenden`, `gönder`) are in it too, unlike English `send`: on those markets the bare verb is the final button and a form-less ATS renders it outside any `<form>`. `IN_FORM_RX` holds the apply verbs (`apply`, `postuler`, `candidati`, `solliciteren`, `aplikuj`, `bewerben`) plus bare `send`: outside a form they are the entry point the loop exists to click, inside a form the same word is the final button. That tier plus `role="button"` is what closes `<div role="button">Send</div>`, the shape a custom ATS ships as its final button. Boundaries are `\p{L}` classes under the `u` flag rather than `\b`, because JS word boundaries are ASCII-only and `/\bотправить\b/` matches nothing. Japanese, Chinese and Korean get the same two tiers through a second matcher with no boundaries at all (`CJK_SUBMIT_RX`: 提出, 送信, 提交, 发送, 發送, 제출, 보내기; `CJK_IN_FORM_RX`: 応募, 申请, 申請, 投递, 投遞, 지원), because those scripts put no spaces around words and a term inside a real label (応募内容を送信する, 提交申请, 제출하기) has letters on both sides. The documented cost of substring matching: 已提交 ("submitted", a tab of sent applications) and 发送验证码 ("send a verification code") are refused anywhere, 応募者一覧 and 지원자 inside a form; a lost click, never a lost application.

**`web/src/lib/apply/drive.ts`**

- Node builds the snapshot lines from element facts, so a submit control is listed **without a ref**: `[--] submit "Label" (not actionable, the human submits)`. The planner can see it exists and stop, but has no name for it. This is what the empty `if (tag === "input" && [...].includes(itype)) { }` block was meant to do; it had no `continue`, so every submit input got a ref like anything else.
- **Refs are per snapshot.** Every existing `data-co-ref` is stripped before the new ones are written, and each turn mints its own random prefix (`e4k2p9-3`), so a ref from turn N names nothing in turn N+1. Numbering from `e0` every turn handed the same name to a different element after a re-render, and both the actionable set and `querySelector` accepted it.
- **The checked node is the acted-on node.** The action path resolves one `ElementHandle`, reads its facts through that handle, and performs the action on that same handle; a `Locator` re-resolves at use time and could hand the action a different node than the one that was cleared. The handle is released in a `finally`. Every action naming a ref is cleared this way, not only `click`: `fill()` on a button throws into a fallback that clicks, so a `type` action on a withheld ref would have pressed it.
- **One facts extractor**, `factsOf`, defined once at top level and handed to Playwright as the `pageFunction` for both call sites, per the first rule in `web/AGENTS.md`. The snapshot pass reads one handle per candidate; with the 70-ref cap saturated that costs 31 to 42 ms per snapshot against a 700 ms settle.
- **Refusals are told apart.** `DriveStep` gained an optional `refusal: "submit-control" | "not-offered" | "gone"` next to the human note. `DrivePanel` renders `submit-control` with a shield icon; the other two keep the amber, since those are the page moving under us, not the guard protecting anything.
- The `click`, `type` fallback and `selectOption` calls now share one explicit 6 s timeout instead of falling through to the context default.

## Tests

`web/tests/lib/submit-guard.test.mjs`, 48 tests in the Given/When/Then style of `apply-exit.test.mjs`. Refused: `input[type=submit]`; an icon-only button named only by aria-label; German, French and Russian submit labels; a submit label hidden behind a generic aria-label; a typeless button inside a form; unlabelled `<button>`, `<a>`, `<div role="button">` and `<input type="button">` inside a form; `<div role="button">Send</div>` inside a form; a control named only by `title`, an icon named only by `alt`, a name that exists only through `aria-labelledby`; an `input[type="button"]` whose `aria` and `value` disagree. Allowed: a link; a declared `type="button"`; a text input; a labelled icon button inside a form; a checkbox reading "Send me updates"; a named `<div role="button">` inside a form; "Send to a friend" outside a form; outside a form, a typeless, an unlabelled button and an unlabelled anchor; a control whose only submit-looking string is its `name`; a fillable field whose placeholder shares a word with the vocabulary. Plus label-parity and regex-boundary cases.

- `cd web && npm test`: 528 tests, 528 pass (480 before this PR). `npm run typecheck` and `npm run build`: clean.
- `node test-all.mjs --quick`: 8095 passed, 0 failed (the web suite counts as one gate there).
- `codex exec review` on the final diff: no findings.

Mutation-checked, one mutant per rule; each row lists every test that goes red and nothing else moves:

| Mutation | Tests that go red |
|---|---|
| drop the typeless-default clause | `a button with no type inside a form is refused, whatever it says` |
| `\b` boundaries, no `u` flag (the old regex shape) | `a Cyrillic submit label is refused`, `«Откликнуться» is refused even outside a form`, `an icon named only by its alt is refused` |
| drop the silence clause | the four unlabelled-inside-a-form tests |
| drop the visible-text source | `a submit label is not hidden by an aria-label that outranks it` |
| drop `role="button"` from the silence clause | `an unlabelled div with role=button inside a form is refused` |
| narrow the silence clause back to button/input | `an unlabelled clickable anchor inside a form is refused, whatever its name says` |
| add `name` into the classification scan | `a control whose only submit-looking string is its name stays allowed`, `an unlabelled anchor outside a form is allowed` |
| drop bare `send` from the in-form tier | `a div with role=button reading "Send" inside a form is refused`, `a native button reading "Send" inside a form is refused`, `IN_FORM_RX: matches the verb` |
| drop `title` and `alt` from the sources | `a control named only by its title is refused`, `an icon named only by its alt is refused` |
| classify on `labelSource`'s single winner again | `a submit term in any label source is refused even when another source outranks it` |
| drop the `isClickable` gate on the wording tier | `a fillable field whose secondary label merely contains a submit word stays allowed` |
| CJK matcher built with word boundaries like the Latin one | the four CJK tests (`the Japanese, Chinese and Korean submit terms are refused inside a real label`, `CJK_SUBMIT_RX: matches as a substring`, ...) |
| `başvur` moved to the in-form tier | `Turkish «Başvur» is refused even outside a form` |
| nl/tr bare send verbs moved to the in-form tier | `the bare send verb of nl, tr, ja, zh and ko is refused even outside a form` |

No `web/tests/**` test uses Playwright today, so the browser-level behaviour was exercised through a throwaway harness that imports `drive.ts` with its four app imports stubbed (a fake CLI that replies with a scripted action), against inline HTML fixtures:

| Fixture | Result |
|---|---|
| `<div role="button">Send</div>` inside a form | listed as `[--] submit "Send"`, absent from the actionable set; a planner that guesses its ref and asks to `type` is refused `not-offered`; the form did not submit |
| re-render, then the planner replays turn 1's ref | turn 2 shares 0 refs with turn 1, 0 elements still carry a turn-1 ref, the replayed ref is refused `not-offered` |
| the same with the per-snapshot prefix removed (mutant) | turn 1's `e2` names a different element in turn 2 and the loop clicks it: the bug the prefix closes |
| a label that becomes "Submit application" 150 ms after the snapshot | refused `submit-control` on the handle re-check |
| a control removed after the snapshot | refused `gone` |
| `type` on an allowed non-fillable control | `fill()` fails and the fallback clicks the element that was checked, not a re-resolved one |

The same harness loaded six public job pages read-only (no clicks, no account) and printed the snapshot the planner would be given:

| Page | Withheld, no ref | Apply entry still actionable |
|---|---|---|
| Greenhouse job board posting | `Submit application` | yes, `button "Apply"` |
| Lever posting | none (the form is a page later) | yes, `link "Apply"` |
| Ashby posting, two companies | none | yes, `link "Apply for this Job"` |
| LinkedIn public job view | `Search` | yes, `button "Apply"` |
| hh.ru search results | `Найти`, `Откликнуться` x2 | no, by design: on hh.ru that click sends the response |

`Search` and `Найти` are the known cost of the rule, not a miss: both really are the submit button of a `<form>`, so the loop cannot use a site's own search box, one refused click on a page it reaches by URL anyway.

## Out of scope

The planner prompt is unchanged except its three example refs, which now read `<ref>` because refs carry a per-turn prefix. A labelled `input[type=reset]` stays clickable; an unlabelled one inside a form is now refused, a side effect of holding the silence rule to `isClickable` rather than a second carve-out. The loop still reads only the main frame and does not walk shadow roots. What no DOM heuristic closes: an allowed control whose own click handler calls `requestSubmit()`. A browser-level interlock on the `submit` event would make the wording tiers a second line of defence rather than the first, and is the natural next step, as a separate change.

## Summary

The apply drive loop no longer exposes submit controls as actionable refs.

Users can still see submit controls in snapshots, but the agent cannot click, type into, or select them. A human must submit the application.

### Changes

- Added shared submit detection in `web/src/lib/apply/submit-guard.mjs:1`.
- Added consistent label extraction and multilingual, Unicode-aware, and CJK matching.
- Excluded hidden and submit controls from actionable refs in `web/src/lib/apply/drive.ts`.
- Revalidated every ref-based action against the live element before creating a locator.
- Added epoch-scoped refs and refusal reasons: `submit-control`, `not-offered`, and `gone`.
- Updated refusal display in `web/src/components/apply-view.tsx`.
- Added 48 submit-guard tests in `web/tests/lib/submit-guard.test.mjs`.

### System files

No changes are reported for `AGENTS.md`, `modes/`, `update-system.mjs`, `DATA_CONTRACT.md`, `providers/`, or `.github/`.

### CODE REVIEW

File: web/src/lib/apply/submit-guard.mjs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+ * - `role`         explicit role, else the tag (`a` reads as "link")
+ * - `inForm`       the element belongs to a `<form>`
+ * - `aria` `placeholder` `text` `value` `name`  the label sources, raw
+ *
+ * @typedef {{ tag?: string, type?: string, explicitType?: boolean, role?: string,
+ *             inForm?: boolean, aria?: string, placeholder?: string, text?: string,
+ *             value?: string, name?: string }} ElementFacts
+ */
+
+const clean = (s) => (s || "").replace(/\s+/g, " ").trim().slice(0, 80);
```
[the remark this answers is not in the document]
Comment (nikitacometa) [reply web/src/lib/apply/submit-guard.mjs]: Fixed in 7951b723. `isSubmitControl` now classifies on the whitespace-collapsed label with no cap (`collapse(labelSource(facts))`), and `snapshotLabel` keeps the 80-char cap for the snapshot line only. Regression test: `a submit term past the 80-char snapshot cap is still refused` uses an aria-only label whose submit wording sits past the cap; with the cap restored in the classifier that test alone goes red. The in-page extractor still caps `textContent` at 200 chars so a `[role=button]` wrapping a section does not cross the wire in full every turn; a control whose accessible text runs past 200 characters before the submit term is a section, not a button.

File: web/src/lib/apply/submit-guard.mjs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  const tag = (facts.tag || "").toLowerCase();
+  const type = (facts.type || "").toLowerCase();
+  if (tag === "input" && (type === "submit" || type === "image")) return true;
+  if (tag === "button" && (type === "submit" || (!facts.explicitType && facts.inForm))) return true;
+  const label = snapshotLabel(facts);
+  // The visible text is tested even when an aria-label outranks it: markup that
+  // labels a button reading "Submit application" as "Continue" would otherwise
+  // hide the wording from the guard while still showing it to the user.
+  if (SUBMIT_RX.test(label) || SUBMIT_RX.test(clean(facts.text))) return true;
+  if (!label && tag === "button" && facts.inForm) return true;
```
[the remark this answers is not in the document]
Comment (nikitacometa) [reply web/src/lib/apply/submit-guard.mjs]: Fixed in 7951b723: the empty-label rule now covers `input[type="button"]` inside a form as well as `<button>`, with the test `an unlabelled input button inside a form is refused` (dropping the input clause turns exactly that test red).

File: web/src/lib/apply/submit-guard.mjs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+/**
+ * The label the planner is shown for an element, and therefore a string the
+ * guard is obliged to test.
+ *
+ * @param {ElementFacts} [facts]
+ */
+export function snapshotLabel(facts = {}) {
+  return clean(facts.aria || facts.placeholder || facts.text || facts.value || facts.name);
+}
```
[the remark this answers is not in the document]
Comment (nikitacometa) [reply web/src/lib/apply/submit-guard.mjs:68]: Declining this one. The guideline at `CONTRIBUTING.md:186-191` is about test files: web suites live under `web/tests/` and must be kept out of `web/src/` because Next.js scans that tree. The example it gives for the tested module is `src/lib/clean-chips.mjs`, i.e. a plain `.mjs` module living inside `web/src/lib/`, which is exactly where this one goes. `web/src/lib/` already holds 38 such modules at fad070fe, including the four apply siblings this follows (`exit.mjs`, `cv-match.mjs`, `cv-selection.mjs`, `answer-prompt.mjs`), and `tests/web-test-layout.test.mjs` enforces that layout on every PR; `drive.ts` imports the module the same way `apply-view.tsx` imports `exit.mjs`.

File: web/src/lib/apply/submit-guard.mjs [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+  if (tag === "input" && (type === "submit" || type === "image")) return true;
+  if (tag === "button" && (type === "submit" || (!facts.explicitType && facts.inForm))) return true;
+  // Classified on the full label, not the 80-char snapshot line: a scripted
+  // control whose submit wording sits past the display cap still submits.
+  const label = collapse(labelSource(facts));
+  // The visible text is tested even when an aria-label outranks it: markup that
+  // labels a button reading "Submit application" as "Continue" would otherwise
+  // hide the wording from the guard while still showing it to the user.
+  const text = collapse(facts.text);
+  if (SUBMIT_RX.test(label) || SUBMIT_RX.test(text)) return true;
```
[the remark this answers is not in the document]
Comment (nikitacometa) [reply web/src/lib/apply/submit-guard.mjs]: Fixed in 49b37dd0. `isSubmitControl` now reads every label source on its own, deduplicated (`names()` in `web/src/lib/apply/submit-guard.mjs`: aria, placeholder, text, value, title, alt), instead of only the single string `labelSource` picks for the planner's line. Your shape, an `input[type="button"]` with `aria="Continue"` and `value="Submit application"`, is refused: test `a submit term in any label source is refused even when another source outranks it`.

`name` is deliberately kept out of that scan rather than added: it is a machine identifier, so a field `name="submit_note"` names a text box, not a submit action. It stays only in the planner-facing `snapshotLabel` fallback (test `a control whose only submit-looking string is its name stays allowed`). The one shape that exclusion left open, an anchor inside a form with no readable name at all, is closed by the silence rule now covering every clickable (`isClickable`), not by letting `name` back in (test `an unlabelled clickable anchor inside a form is refused, whatever its name says`).

File: web/src/lib/apply/submit-guard.mjs [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+  // control whose submit wording sits past the display cap still submits.
+  const label = collapse(labelSource(facts));
+  // The visible text is tested even when an aria-label outranks it: markup that
+  // labels a button reading "Submit application" as "Continue" would otherwise
+  // hide the wording from the guard while still showing it to the user.
+  if (SUBMIT_RX.test(label) || SUBMIT_RX.test(collapse(facts.text))) return true;
+  // No wording at all inside a form: an icon-only <button>, or an
+  // <input type="button"> that submits through its handler. Refused rather
+  // than guessed at.
+  if (!label && facts.inForm && (tag === "button" || (tag === "input" && type === "button"))) return true;
```
[the remark this answers is not in the document]
Comment (nikitacometa) [reply web/src/lib/apply/submit-guard.mjs]: Covered in 49b37dd0. The silence rule reads `role === "button"` as a button (and, since this round, any clickable via `isClickable`), so an unlabelled `<div role="button">` inside a form is refused: test `an unlabelled div with role=button inside a form is refused`, plus the anchor case above. Mutation-checked: dropping `role === "button"` from the clause, or narrowing `isClickable` back to button/input only, turns exactly those tests red.

**Review by santifer — CHANGES_REQUESTED** [review]
@nikitacometa — I came in with five holes from the issue and my own read of `main`. **You have already closed four of them**, and the architecture you chose is the one I would have asked for, so let me be specific about what I checked rather than wave at it.

**Closed, verified by running your module and reading `drive.ts` at your head:**

- *Two derivations of the same label.* `factsOf` reads the element once and both sides go through `snapshotLabel(el)` / `isSubmitControl(el)`. One rule, one place. `snapshotLabel({aria:"Submit application", text:""})` → `"Submit application"`, and `isSubmitControl` on it → `true`. The icon-only case is gone.
- *`innerText` vs `textContent`.* Moot now — there is no second string to disagree with.
- *The empty `if` in main.* `main` had `if (tag === "input" && [...].includes(itype)) { /* comment only */ }` — a block that promised submits were read-only context and did nothing. Yours emits them as `[--] submit "…" (not actionable, the human submits)` and, more importantly, never adds them to `actionable`. The promise is now code.
- *Fail-open on a read error.* `main` did `catch(() => "")`, and `""` never matches the regex, so a failed read meant **click**. Yours: `const target = now && !isSubmitControl(now) ? handle : null` — unreadable is refused. Your comment says it outright, which is why it will survive the next edit.

The per-turn `epoch` on the refs and the three distinguished refusals weren't on my list at all. The second one is a product judgement I'd have missed: a guard that calls a vanished element "a submit button" teaches people to ignore the message that matters.

## The one thing left, and it needs more than a longer list

`SUBMIT_TERMS` doesn't cover seven of the project's own locales. Measured against your `SUBMIT_RX`:

```
true   Absenden · Wyślij · Отправить · Invia candidatura
false  Postuler · Versturen · 送信 · 제출 · Gönder · 提交
```

The project ships evaluation modes in fr, nl, ja, ko, tr, zh and zh-TW, so those are real buttons real users will face.

**And adding the words is not enough for CJK** — this is the part worth catching before you write it. Your `wordRx` builds `(^|[^\p{L}])(term)($|[^\p{L}])`, which is right for space-separated languages. Japanese, Chinese and Korean don't put spaces around words, so a term inside a real button label has letters on both sides and the boundary never matches:

```
with the CJK terms added to your own wordRx:
  true   送信                 the bare term
  false  応募内容を送信する      the same term in a real button label
  false  提交申请
  false  제출하기
  true   Başvuruyu Gönder     (space-separated: works fine)
  true   Postuler maintenant
```

So `fr`, `nl` and `tr` are a one-line addition to the array. `ja`, `ko`, `zh` need a second matcher without word boundaries — substring is correct there, and the false-positive risk is much lower than for a short Latin word.

That's the only change I'm asking for. Everything else here I'd merge as it stands, and thank you for finding it: a guard that reads a different string from the one the planner sees is the exact shape of bug this codebase keeps producing, and you found it on the one line where it costs a person an application they didn't choose to send.
